src/bootstrap/src/core/build_steps/dist.rs

```diff
@@ -2392,6 +2393,45 @@ impl Step for BuildManifest {
     }
 }
 
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub struct Enzyme {
```

Comment:
If rustc dynamically links Enzyme rather than dlopen's it at runtime, that means that Enzyme needs to be unconditionally present when running rustc if Enzyme support is enabled. As such I think Enzyme will have to be put in the same component as rustc.

Reply:
I am not testing dist builds, so this code section was best-effort only.
But if you look at https://github.com/rust-lang/rust/pull/129175, especially
https://github.com/rust-lang/rust/blob/e1eaacf1909127b243dc0506972e40705955fcf8/compiler/rustc_codegen_llvm/src/llvm/enzyme_ffi.rs#L250, you'll see that I have backends available, so we won't link against it if it isn't build.

Reply:
> so we won't link against it if it isn't build.

But if we do link against it, it needs to be available at runtime. So my suggestion would be to put the Enzyme dylib in the rustc component of and only if Enzyme support is enabled at all.

Reply:
Sounds good, what would I need to do to have it as part of rustc? Also, what about static linking, some contributors wanted to set that up, would there be any issues with it?

Reply:
I think libEnzyme is already included in the rustc component and the enzyme component doesn't contain anything other than the readme and license. As such try removing this dist step and the enzyme mention in tool.rs. (The reference in tool.rs only works for building cargo projects, which the Enzyme repo doesn't contain.)

Reply:
I then also had to revert utils/tarball.rs, but I had also only added these three for upstreaming since I thought they'd be necessary, so I'm happy to drop them. Can I mark this as resolved?